### PR TITLE
exec: add memory grow support

### PR DESCRIPTION
**Pull request by chenfengjin**
## Description

What is the purpose of the change?
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Brief of your solution

Please describe your solution to solve the issue or feature request.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

### CODE REVIEW

File: exec/xvm.c [hunk 1]
```diff
@@ -126,18 +145,61 @@ static void wasm_rt_free_memory(wasm_rt_memory_t* mem) {
...
+        wasm_rt_trap(WASM_RT_TRAP_OOB);
+    }
+    if (delta < 0) {
+        wasm_rt_trap(WASM_RT_TRAP_OOB);
+    }
 
     uint32_t old_pages = memory->pages;
     uint32_t new_pages = memory->pages + delta;
+    if (new_pages > 65535) {
+        return -1;
```
Comment (wangchao0222) [inline exec/xvm.c:160]: 返回值类型是uint32_t，这里返回负值会不会有问题？

File: exec/xvm.h [hunk 2]
```diff
@@ -44,9 +43,30 @@ typedef struct xvm_resolver_t {
 } xvm_resolver_t;
 
 struct FuncType;
+// TODO
+typedef int bool;
```
Comment (wangchao0222) [inline exec/xvm.h:47]: 46号的TODO没有具体描述，或者是否已经解决了
已经自定义了C语言的布尔类型，建议直接定义true和false的宏定义

Comment (chenfengjin) [reply exec/xvm.h:46]: resolved

File: exec/xvm.c (hunk 1 above)
Comment (chenfengjin) [reply exec/xvm.c:160]: uint32_t 返回 -1 会被解析成 0xFFFFFFFF,  其实和返回 2^32-1 是一样的

**Review by wangchao0222 — APPROVED** [review]

**Review by godeamon — APPROVED** [review]

**Pull request merged**